native/components/message-markdown-rules.react.js

```diff
@@ -0,0 +1,155 @@
+// @flow
+
+import * as React from 'react';
+import { Text } from 'react-native';
+import SimpleMarkdown from 'simple-markdown';
+
+type State = { [string]: any };
+
+type ASTNode = SingleASTNode | Array<SingleASTNode>;
+
+type SingleASTNode = {
+  type: string,
+  [string]: any,
+};
+
+type ReactElement = React$Element<any>;
+
+type Output<Result> = (node: ASTNode, state?: ?State) => Result;
+
+export default function(styles: any, opts: any = {}) {
+  const LINK_INSIDE = '(?:\\[[^\\]]*\\]|[^\\]]|\\](?=[^\\[]*\\]))*';
+  const LINK_HREF_AND_TITLE =
+    '\\s*<?([^\\s]*?)>?(?:\\s+[\'"]([\\s\\S]*?)[\'"])?\\s*';
+  var pressHandler = function(target) {
+    if (opts.onLink) {
+      opts.onLink(target).catch(function(error) {
+        console.log(
+          'There has been a problem with this action. ' + error.message,
+        );
+        throw error;
+      });
+    }
+  };
+  return {
+    autolink: {
+      react: function(
+        node: SingleASTNode,
+        output: Output<ReactElement>,
+        state: State,
+      ) {
+        state.withinText = true;
+        const _pressHandler = () => {
+          pressHandler(node.target);
+        };
+        const autolink = React.createElement(
+          Text,
+          {
+            key: state.key,
+            style: styles.autolink,
+            onPress: _pressHandler,
+          },
+          output(node.content, state),
+        );
+        return autolink;
+      },
+    },
+    link: {
```

Comment:
Do we need `link` for `autolink` to work? If not, let's get rid of it

Reply:
simple-markdown uses `link` for `autolink`, so we need to have it

Reply:
Ah, I see... looking at the code now. `autolink`, `mailto`, and `url` all go through `link`.